[WPE] Upstream the new WPE platform API

carlosgcampos opened the pull request:
#### b12e7ed2ad3a47ab322507ebac214225f2298acc
<pre>
[WPE] Upstream the new WPE platform API
<a href="https://bugs.webkit.org/show_bug.cgi?id=265286">https://bugs.webkit.org/show_bug.cgi?id=265286</a>

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

New WPE platform API is still work in progress, but we have enough
already to know that this is the way we want to go, so we can continue
working upstream.

The new API is always built, but the old one will be used by default, so
there shouldn&apos;t be any change in behavior. MiniBrowser and WKTR can use
the new api by using a new command line option to enable it.

* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::pixelsPerLineStep):
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::createPlatformDisplay):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
(WebCore::PlatformDisplay::setUseDMABufForRendering):
* Source/WebCore/platform/libwpe/PlatformKeyboardEventLibWPE.cpp:
(WebCore::PlatformKeyboardEvent::currentStateOfModifierKeys):
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/Shared/NativeWebKeyboardEvent.h:
* Source/WebKit/Shared/NativeWebMouseEvent.h:
* Source/WebKit/Shared/NativeWebTouchEvent.h:
* Source/WebKit/Shared/NativeWebWheelEvent.h:
* Source/WebKit/Shared/WebKeyboardEvent.h:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.serialization.in:
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/Shared/libwpe/WebEventFactory.h:
* Source/WebKit/Shared/wpe/GRefPtrWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WTF::refGPtr):
(WTF::derefGPtr):
* Source/WebKit/Shared/wpe/GRefPtrWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
* Source/WebKit/Shared/wpe/NativeWebKeyboardEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebKeyboardEvent::NativeWebKeyboardEvent):
* Source/WebKit/Shared/wpe/NativeWebMouseEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebMouseEvent::NativeWebMouseEvent):
* Source/WebKit/Shared/wpe/NativeWebTouchEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebTouchEvent::NativeWebTouchEvent):
* Source/WebKit/Shared/wpe/NativeWebWheelEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebWheelEvent::NativeWebWheelEvent):
* Source/WebKit/Shared/wpe/WebEventFactoryWPE.cpp: Added.
(WebKit::wallTimeForEvent):
(WebKit::modifiersFromWPEModifiers):
(WebKit::buttonForWPEButton):
(WebKit::buttonFromWPEModifiers):
(WebKit::movementDeltaFromEvent):
(WebKit::pressedMouseButtons):
(WebKit::positionFromEvent):
(WebKit::WebEventFactory::createWebMouseEvent):
(WebKit::WebEventFactory::createWebWheelEvent):
(WebKit::WebEventFactory::createWebKeyboardEvent):
(WebKit::WebEventFactory::createWebTouchEvent):
* Source/WebKit/Shared/wpe/WebKeyboardEventWPE.cpp: Copied from Source/WebCore/platform/libwpe/PlatformKeyboardEventLibWPE.cpp.
(WebKit::WebKeyboardEvent::keyValueStringForWPEKeyval):
(WebKit::WebKeyboardEvent::keyCodeStringForWPEKeycode):
(WebKit::WebKeyboardEvent::keyIdentifierForWPEKeyval):
(WebKit::WebKeyboardEvent::windowsKeyCodeForWPEKeyval):
(WebKit::WebKeyboardEvent::singleCharacterStringForWPEKeyval):
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/C/wpe/WKView.cpp:
(WKViewCreate):
(WKViewCreateDeprecated):
(WKViewGetView):
(WKViewSetViewClient): Deleted.
* Source/WebKit/UIProcess/API/C/wpe/WKView.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(WebKitWebViewClient::enterFullScreen):
(WebKitWebViewClient::exitFullScreen):
(webkitWebViewConstructed):
(webkitWebViewSetProperty):
(webkitWebViewGetProperty):
(webkit_web_view_class_init):
(webkitWebViewCreatePage):
(webkit_web_view_get_display):
(webkit_web_view_get_wpe_view):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in:
* Source/WebKit/UIProcess/API/wpe/APIViewClient.h:
(API::ViewClient::enterFullScreen):
(API::ViewClient::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::wpeView const):
(WebKit::PageClientImpl::hostFileDescriptor):
(WebKit::PageClientImpl::setCursor):
(WebKit::PageClientImpl::setCursorHiddenUntilMouseMoves):
(WebKit::PageClientImpl::enterAcceleratedCompositingMode):
(WebKit::PageClientImpl::exitAcceleratedCompositingMode):
(WebKit::PageClientImpl::updateAcceleratedCompositingMode):
(WebKit::PageClientImpl::enterFullScreen):
(WebKit::PageClientImpl::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::View::View):
(WKWPE::m_wpeDisplay):
(WKWPE::View::~View):
(WKWPE::View::isFullScreen const):
(WKWPE::View::willEnterFullScreen):
(WKWPE::View::enterFullScreen):
(WKWPE::View::didEnterFullScreen):
(WKWPE::View::willExitFullScreen):
(WKWPE::View::exitFullScreen):
(WKWPE::View::didExitFullScreen):
(WKWPE::View::requestExitFullScreen):
(WKWPE::View::setFullScreen):
(WKWPE::View::updateAcceleratedSurface):
(WKWPE::View::touchPointsForEvent):
(WKWPE::View::setCursor):
(WKWPE::m_backend): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
(WKWPE::View::create):
(WKWPE::View::wpeView const):
(WKWPE::View::isFullScreen): Deleted.
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewClient.h:
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp:
(webkit_web_view_new):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
(WebKit::WebPageProxy::updateCurrentModifierState):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
(WebKit::DisplayVBlankMonitorDRM::create):
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.cpp: Added.
(WebKit::AcceleratedBackingStoreDMABuf::create):
(WebKit::AcceleratedBackingStoreDMABuf::AcceleratedBackingStoreDMABuf):
(WebKit::AcceleratedBackingStoreDMABuf::~AcceleratedBackingStoreDMABuf):
(WebKit::AcceleratedBackingStoreDMABuf::updateSurfaceID):
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBuffer):
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBufferSHM):
(WebKit::AcceleratedBackingStoreDMABuf::didDestroyBuffer):
(WebKit::AcceleratedBackingStoreDMABuf::frame):
(WebKit::AcceleratedBackingStoreDMABuf::frameDone):
(WebKit::AcceleratedBackingStoreDMABuf::bufferRendered):
* Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.h: Added.
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::wpeView const):
(WebKit::WebPageProxy::preferredBufferFormats const):
(WebKit::WebPageProxy::preferredBufferFormatsDidChange):
(WebKit::WebPageProxy::currentStateOfModifierKeys):
* Source/WebKit/WPEPlatform: Added.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.h:
(WebKit::AcceleratedSurface::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::SwapChain::setupBufferFormat):
(WebKit::AcceleratedSurfaceDMABuf::SwapChain::nextTarget):
(WebKit::AcceleratedSurfaceDMABuf::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Source/WebKit/WebProcess/WebPage/wpe/WebPageWPE.cpp:
(WebKit::WebPage::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
* Source/WebKit/wpe/wpe-webkit-uninstalled.pc.in:
* Source/WebKit/wpe/wpe-webkit.pc.in:
* Source/cmake/FindLibInput.cmake: Added.
* Source/cmake/FindUdev.cmake: Added.
* Source/cmake/OptionsWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/MiniBrowser/wpe/main.cpp:
(createViewBackend):
(createWebView):
(activate):
(main):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.cmd_line):
* Tools/Scripts/webkitpy/port/wpe.py:
(WPEPort.setup_environ_for_server):
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_identifier_name_in_declaration):
* Tools/TestWebKitAPI/wpe/PlatformWebViewWPE.cpp:
(TestWebKitAPI::PlatformWebView::initialize):
* Tools/WebKitTestRunner/EventSenderProxy.h:
* Tools/WebKitTestRunner/Options.cpp:
(WTR::handleOptionWPEPlatformAPI):
(WTR::OptionsHandler::OptionsHandler):
* Tools/WebKitTestRunner/Options.h:
* Tools/WebKitTestRunner/PlatformWPE.cmake:
* Tools/WebKitTestRunner/PlatformWebView.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::initialize):
* Tools/WebKitTestRunner/TestController.h:
(WTR::TestController::useWPEPlatformAPI const):
* Tools/WebKitTestRunner/libwpe/EventSenderProxyClient.h: Copied from Source/WebKit/UIProcess/API/wpe/APIViewClient.h.
(WTR::EventSenderProxyClient::addTouchPoint):
(WTR::EventSenderProxyClient::updateTouchPoint):
(WTR::EventSenderProxyClient::touchStart):
(WTR::EventSenderProxyClient::touchMove):
(WTR::EventSenderProxyClient::touchEnd):
(WTR::EventSenderProxyClient::touchCancel):
(WTR::EventSenderProxyClient::clearTouchPoints):
(WTR::EventSenderProxyClient::releaseTouchPoint):
(WTR::EventSenderProxyClient::cancelTouchPoint):
(WTR::EventSenderProxyClient::EventSenderProxyClient):
* Tools/WebKitTestRunner/libwpe/EventSenderProxyClientLibWPE.cpp: Copied from Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp.
(WTR::EventSenderProxyClientLibWPE::EventSenderProxyClientLibWPE):
(WTR::viewBackend):
(WTR::secToMsTimestamp):
(WTR::senderButtonToWPEButton):
(WTR::modifierForButton):
(WTR::wkEventModifiersToWPE):
(WTR::EventSenderProxyClientLibWPE::mouseDown):
(WTR::EventSenderProxyClientLibWPE::mouseUp):
(WTR::EventSenderProxyClientLibWPE::mouseMoveTo):
(WTR::EventSenderProxyClientLibWPE::mouseScrollBy):
(WTR::wpeKeySymForKeyRef):
(WTR::EventSenderProxyClientLibWPE::keyDown):
(WTR::EventSenderProxyClientLibWPE::addTouchPoint):
(WTR::EventSenderProxyClientLibWPE::updateTouchPoint):
(WTR::EventSenderProxyClientLibWPE::touchStart):
(WTR::EventSenderProxyClientLibWPE::touchMove):
(WTR::EventSenderProxyClientLibWPE::touchEnd):
(WTR::EventSenderProxyClientLibWPE::touchCancel):
(WTR::EventSenderProxyClientLibWPE::clearTouchPoints):
(WTR::EventSenderProxyClientLibWPE::releaseTouchPoint):
(WTR::EventSenderProxyClientLibWPE::cancelTouchPoint):
(WTR::EventSenderProxyClientLibWPE::getUpdatedTouchEvents):
(WTR::EventSenderProxyClientLibWPE::removeUpdatedTouchEvents):
(WTR::EventSenderProxyClientLibWPE::prepareAndDispatchTouchEvent):
* Tools/WebKitTestRunner/libwpe/EventSenderProxyClientLibWPE.h: Added.
* Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp:
(WTR::EventSenderProxy::EventSenderProxy):
(WTR::EventSenderProxy::mouseDown):
(WTR::EventSenderProxy::mouseUp):
(WTR::EventSenderProxy::mouseMoveTo):
(WTR::EventSenderProxy::mouseScrollBy):
(WTR::EventSenderProxy::keyDown):
(WTR::EventSenderProxy::addTouchPoint):
(WTR::EventSenderProxy::updateTouchPoint):
(WTR::EventSenderProxy::touchStart):
(WTR::EventSenderProxy::touchMove):
(WTR::EventSenderProxy::touchEnd):
(WTR::EventSenderProxy::touchCancel):
(WTR::EventSenderProxy::clearTouchPoints):
(WTR::EventSenderProxy::releaseTouchPoint):
(WTR::EventSenderProxy::cancelTouchPoint):
(WTR::viewBackend): Deleted.
(WTR::secToMsTimestamp): Deleted.
(WTR::senderButtonToWPEButton): Deleted.
(WTR::modifierForButton): Deleted.
(WTR::wkEventModifiersToWPE): Deleted.
(WTR::wpeKeySymForKeyRef): Deleted.
(WTR::EventSenderProxy::getUpdatedTouchEvents): Deleted.
(WTR::EventSenderProxy::removeUpdatedTouchEvents): Deleted.
(WTR::EventSenderProxy::prepareAndDispatchTouchEvent): Deleted.
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClient.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WTR::PlatformWebViewClient::view const):
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClientLibWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.cpp.
(WTR::PlatformWebViewClientLibWPE::PlatformWebViewClientLibWPE):
(WTR::PlatformWebViewClientLibWPE::~PlatformWebViewClientLibWPE):
(WTR::PlatformWebViewClientLibWPE::addToWindow):
(WTR::PlatformWebViewClientLibWPE::removeFromWindow):
(WTR::PlatformWebViewClientLibWPE::snapshot):
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClientLibWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
* Tools/WebKitTestRunner/libwpe/PlatformWebViewLibWPE.cpp:
(WTR::PlatformWebView::PlatformWebView):
(WTR::PlatformWebView::addToWindow):
(WTR::PlatformWebView::removeFromWindow):
(WTR::PlatformWebView::windowSnapshotImage):
* Tools/WebKitTestRunner/wpe/EventSenderProxyClientWPE.cpp: Added.
(WTR::EventSenderProxyClientWPE::EventSenderProxyClientWPE):
(WTR::secToMsTimestamp):
(WTR::wkEventModifiersToWPE):
(WTR::eventSenderButtonToWPEButton):
(WTR::modifierForButton):
(WTR::EventSenderProxyClientWPE::mouseDown):
(WTR::EventSenderProxyClientWPE::mouseUp):
(WTR::EventSenderProxyClientWPE::mouseMoveTo):
(WTR::EventSenderProxyClientWPE::mouseScrollBy):
(WTR::wpeKeyvalForKeyRef):
(WTR::EventSenderProxyClientWPE::keyDown):
* Tools/WebKitTestRunner/wpe/EventSenderProxyClientWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
* Tools/WebKitTestRunner/wpe/PlatformWebViewClientWPE.cpp: Added.
(WTR::PlatformWebViewClientWPE::PlatformWebViewClientWPE):
(WTR::PlatformWebViewClientWPE::~PlatformWebViewClientWPE):
(WTR::PlatformWebViewClientWPE::addToWindow):
(WTR::PlatformWebViewClientWPE::removeFromWindow):
(WTR::PlatformWebViewClientWPE::snapshot):
* Tools/WebKitTestRunner/wpe/PlatformWebViewClientWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.

Canonical link: <a href="https://commits.webkit.org/271330@main">https://commits.webkit.org/271330@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e21e11dbb7a6b9da7fe5a638f61f2566d241e6f5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/28062 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/6701 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/29381 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/30592 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/25581 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/28558 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/8708 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/4088 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/30592 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/28329 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/8708 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/29381 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/30592 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 webkitpy~~](https://ews-build.webkit.org/#/builders/6/builds/28221 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/8708 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/29381 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/31281 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/24289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/8708 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/29381 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/31281 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/27186 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/4874 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/4088 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/31281 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/6415 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/29381 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/34682 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/5313 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/34682 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3625 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/5372 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebCore/platform/Scrollbar.cpp:
```diff
@@ -60,7 +60,7 @@ void Scrollbar::setShouldUseFixedPixelsPerLineStepForTesting(bool useFixedPixels
 int Scrollbar::pixelsPerLineStep(int viewWidthOrHeight)
 {
-#if PLATFORM(GTK)
+#if PLATFORM(GTK) || PLATFORM(WPE)
```
We've had issues in the past where people reported that mouse wheel scroll was too slow in WPE and at some point we were considering to allow configuring this, and we ended up “solving” in Cog... Nice that we get to reuse code here 😄

carlosgcampos replied:
Yes, with the new api we get the same than the gtk port.

aperezdc commented on Source/WebKit/Shared/wpe/WebEventFactoryWPE.cpp:
```diff
@@ -0,0 +1,232 @@
+static WebMouseEventButton buttonForWPEButton(guint button)
+{
+    switch (button) {
+    case 1:
```
I wonder if we might want to have symbolic names for the mouse buttons. That being said, I think it's okay to use integer literals, the meaning is quite clear.

aperezdc commented on Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
```diff
@@ -86,6 +92,138 @@ View::View(struct wpe_view_backend* backend, const API::PageConfiguration& baseC
+            webView.m_inputMethodFilter.notifyFocusedOut();
+            webView.page().activityStateDidChange(flagsToUpdate);
+        }), this);
+        g_signal_connect(m_wpeView.get(), "state-changed", G_CALLBACK(+[](WPEView* view, WPEViewState previousState, gpointer userData) {
```
This is a value that we want to watch for changes. Wouldn't it make sense that `state` is a property, i.e. `WPEView:state`, and then code here would connect to the `notify::state` signal?

carlosgcampos replied:
I agree, and this is what I planned initially, but in this particular case in which we need to compare with the previous state, it's very convenient that the signal callback includes the previous state. In GTK4 it's a property and we have to keep a copy of the state in webkit to get the difference and know what changed.

aperezdc replied:
I see, that sounds like a reasonable motivation to avoid using a property here 🆗

aperezdc commented on Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
```diff
@@ -86,6 +92,138 @@ View::View(struct wpe_view_backend* backend, const API::PageConfiguration& baseC
+                }
+            }
+        }), this);
+        g_signal_connect(m_wpeView.get(), "preferred-dma-buf-formats-changed", G_CALLBACK(+[](WPEView*, gpointer userData) {
```
Same here, would it make sense for this to be a `WPEView:preferred-dmabuf-formats` property?

carlosgcampos replied:
I this case it probably makes sense, we can change it in a follow up. As soon as we land this I'll create bugs for all the remaining tasks, feel free to create one for this.

aperezdc replied:
We'll have more things to polish anyway so a follow-up patch sounds fine.

aperezdc commented on Source/WebKit/WPEPlatform/CMakeLists.txt:
```diff
@@ -0,0 +1,209 @@
+add_custom_command(
+    OUTPUT ${WPEPlatform_DERIVED_SOURCES_DIR}/wpe/WPEEnumTypes.h
+    DEPENDS ${WPEPlatform_ENUM_GENERATION_HEADERS}
+    COMMAND glib-mkenums --template ${WEBKIT_DIR}/WPEPlatform/wpe/WPEEnumTypes.h.in ${WPEPlatform_ENUM_GENERATION_HEADERS} | sed s/w_pe/wpe/ | sed s/WPE_TYPE_PE/WPE_TYPE/ | sed s/WPE_TYPEEGL/WPE_TYPE_EGL/ | sed s/wpeegl/wpe_egl/ > ${WPEPlatform_DERIVED_SOURCES_DIR}/wpe/WPEEnumTypes.h
```
Just thinking out loud: At some point in the future I would like to factor out all the manual `glib-mkenums` rules into a CMake macro or function, to avoid error-prone copypasta.

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
```diff
@@ -0,0 +1,371 @@
+ * @strides: the buffer strides
+ * @modifier: the buffer modifier
+ *
+ * Crerate a new #WPEBufferDMABuf for the given parameters.
```
Typo:

```suggestion
 * Create a new #WPEBufferDMABuf for the given parameters.
```

aperezdc reviewed (approved):
This is quite big that I could not read the whole patch in detail... and it doesn't make sense, because some parts only move code around or make it conditional due to the display vs. backend choice. So I hope I haven't missed anything fundamental.

My main comment would be that there are a few cases in which we have signals like `WPEView::state-changed`, and I wonder if it would make more sense to make those into properties and have the code that uses them connect to e.g. `notify::state` instead, which seems to be more fitting with the idea of “properties are values that may be monitored for changes” while signals “tell when something happened”. Is there some particular reason for the choice? Performance?

Otherwise, I like how this looks. Very neat work 💪🏼

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
```diff
@@ -785,6 +787,11 @@ static void webkitWebViewConstructed(GObject* object)
     if (priv->relatedView) {
+#if PLATFORM(WPE)
+        if (priv->display)
+            g_critical("WebKitWebView display property can't be set when releated-view is set too, passed display value is ignored.");
```
```suggestion
            g_critical("WebKitWebView display property can't be set when related-view is set too, passed display value is ignored.");
```

mcatanzaro commented on Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
```diff
@@ -115,27 +130,40 @@ class View : public API::ObjectImpl<API::Object::Type::View> {
+    std::unique_ptr<WebKit::AcceleratedBackingStoreDMABuf> m_backingStore;
 
     struct wpe_view_backend* m_backend;
+    WPEDisplay* m_wpeDisplay { nullptr };
```
I don't like this. Who owns the `WPEDisplay*`? It's a refcounted GObject, so we might as well use `GRefPtr`. If that would cause a reference cycle, then we can use `GWeakPtr`.

(It'd be good to add a comment to document ownership of the `wpe_view_backend` too. Storing member variables as raw pointers is historically a source of many security bugs for us.)

carlosgcampos replied:
The display is owned by the WPEWebView owners (WebKitWebView for glib API, PlatformWebViewClientWPE for C API). WPEWebView is always created with unique_ptr so it's always deleted before its owner. I don't think it makes sense to use a weak reference, because it can't be deleted while the WPEWebView is alive, but I don't mind to use GRefPtr, because even though it's useless, there's no risk of cycles.

carlosgcampos replied:
hmm, but we don't really need to keep the display, since we can get if from the view, so I'm going to remove it.

mcatanzaro commented on Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.h:
```diff
@@ -0,0 +1,85 @@
+    void bufferRendered();
+
+    WebPageProxy& m_webPage;
+    WPEView* m_wpeView { nullptr };
```
Use `CheckedPtr<WebPageProxy>` for `m_webPage`, because `WebPageProxy` already uses `CanMakeCheckedPtr`.

Use `GRefPtr` or `GWeakPtr` for `m_wpeView`.

carlosgcampos replied:
Same here, the view is owned by WPEWebView, which is the owner of AcceleratedBackingStore that is not refcounted, so it's always going to be deleted before WPEView. We explicitly call `m_backingStore = nullptr;` in the destructor to ensure that. Again, I don't mind to use GRefPtr because there's not going to be cycle, just a useless reference increase.

mcatanzaro commented on Source/WebKit/WPEPlatform/docs/tutorial-browser.md:
```diff
@@ -0,0 +1,4 @@
+Title: How to use WPE platform to create a WPEWebKit browser
```
```suggestion
Title: How to Use WPE Platform to Create a WPE WebKit Browser
```

mcatanzaro commented on Source/WebKit/WPEPlatform/docs/tutorial-platform.md:
```diff
@@ -0,0 +1,4 @@
+Title: How to use WPE platform to extend the platform support
```
```suggestion
Title: How to Use WPE Platform to Extend the Platform Support
```

mcatanzaro commented on Source/WebKit/WPEPlatform/wpe/WPEBuffer.cpp:
```diff
@@ -0,0 +1,233 @@
+            "display",
+            nullptr, nullptr,
+            WPE_TYPE_DISPLAY,
+            static_cast<GParamFlags>(G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS));
```
Since this has moved to WebKit, now you should use `WEBKIT_PARAM_READWRITE` so you don't need to also use `G_PARAM_STATIC_STRINGS` all over the place. I'll only leave this comment once but of course it applies to everywhere.

mcatanzaro commented on Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp:
```diff
@@ -0,0 +1,371 @@
+#include <gbm.h>
+#include <mutex>
+#include <unistd.h>
+#endif
```
Most of these can be included unconditionally, even if they're currently only used in the `USE(GBM)` case. This will reduce future build failures whenever somebody decides to use something from the non-GBM headers in the future. Build failures aren't fun.

I would only conditionally include gbm.h and maybe drm_fourcc.h.

mcatanzaro commented on Source/WebKit/WPEPlatform/wpe/WPEEGLError.h:
```diff
@@ -0,0 +1,54 @@
+#endif
+
+#ifndef WPEEGLError_h
+#define WPEEGLError_h
```
This is not important, but if you have extra time, IMO it's nicer to swap these:

```suggestion
#ifndef WPEEGLError_h
#define WPEEGLError_h

#if !defined(__WPE_PLATFORM_H_INSIDE__) && !defined(BUILDING_WEBKIT)
#error "Only <wpe/wpe-platform.h> can be included directly."
#endif
```

At some point I heard this allows for some sort of optimization. Not sure if it's true. Probably not significant.

mcatanzaro reviewed (approved):
I've only quickly skimmed over the first half of this, but I'm going to say it looks good. Reviewing this in detail is not practical, but it appears to be high quality as usual.

It'd be nice to reduce use of raw pointers wherever possible, but we can improve that in the future. I'll complain only about member variables, since they are riskiest.

It's getting hard to keep track of all the different WPE view classes. Not sure what we can do about this, because "view" is really the right name. Alas.

carlosgcampos committed:
[WPE] Upstream the new WPE platform API
https://bugs.webkit.org/show_bug.cgi?id=265286

Reviewed by Adrian Perez de Castro and Michael Catanzaro.

New WPE platform API is still work in progress, but we have enough
already to know that this is the way we want to go, so we can continue
working upstream.

The new API is always built, but the old one will be used by default, so
there shouldn't be any change in behavior. MiniBrowser and WKTR can use
the new api by using a new command line option to enable it.

* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::pixelsPerLineStep):
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::createPlatformDisplay):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
(WebCore::PlatformDisplay::setUseDMABufForRendering):
* Source/WebCore/platform/libwpe/PlatformKeyboardEventLibWPE.cpp:
(WebCore::PlatformKeyboardEvent::currentStateOfModifierKeys):
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/Shared/NativeWebKeyboardEvent.h:
* Source/WebKit/Shared/NativeWebMouseEvent.h:
* Source/WebKit/Shared/NativeWebTouchEvent.h:
* Source/WebKit/Shared/NativeWebWheelEvent.h:
* Source/WebKit/Shared/WebKeyboardEvent.h:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.serialization.in:
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/Shared/libwpe/WebEventFactory.h:
* Source/WebKit/Shared/wpe/GRefPtrWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WTF::refGPtr):
(WTF::derefGPtr):
* Source/WebKit/Shared/wpe/GRefPtrWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
* Source/WebKit/Shared/wpe/NativeWebKeyboardEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebKeyboardEvent::NativeWebKeyboardEvent):
* Source/WebKit/Shared/wpe/NativeWebMouseEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebMouseEvent::NativeWebMouseEvent):
* Source/WebKit/Shared/wpe/NativeWebTouchEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebTouchEvent::NativeWebTouchEvent):
* Source/WebKit/Shared/wpe/NativeWebWheelEventWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WebKit::NativeWebWheelEvent::NativeWebWheelEvent):
* Source/WebKit/Shared/wpe/WebEventFactoryWPE.cpp: Added.
(WebKit::wallTimeForEvent):
(WebKit::modifiersFromWPEModifiers):
(WebKit::buttonForWPEButton):
(WebKit::buttonFromWPEModifiers):
(WebKit::movementDeltaFromEvent):
(WebKit::pressedMouseButtons):
(WebKit::positionFromEvent):
(WebKit::WebEventFactory::createWebMouseEvent):
(WebKit::WebEventFactory::createWebWheelEvent):
(WebKit::WebEventFactory::createWebKeyboardEvent):
(WebKit::WebEventFactory::createWebTouchEvent):
* Source/WebKit/Shared/wpe/WebKeyboardEventWPE.cpp: Copied from Source/WebCore/platform/libwpe/PlatformKeyboardEventLibWPE.cpp.
(WebKit::WebKeyboardEvent::keyValueStringForWPEKeyval):
(WebKit::WebKeyboardEvent::keyCodeStringForWPEKeycode):
(WebKit::WebKeyboardEvent::keyIdentifierForWPEKeyval):
(WebKit::WebKeyboardEvent::windowsKeyCodeForWPEKeyval):
(WebKit::WebKeyboardEvent::singleCharacterStringForWPEKeyval):
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/C/wpe/WKView.cpp:
(WKViewCreate):
(WKViewCreateDeprecated):
(WKViewGetView):
(WKViewSetViewClient): Deleted.
* Source/WebKit/UIProcess/API/C/wpe/WKView.h:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(WebKitWebViewClient::enterFullScreen):
(WebKitWebViewClient::exitFullScreen):
(webkitWebViewConstructed):
(webkitWebViewSetProperty):
(webkitWebViewGetProperty):
(webkit_web_view_class_init):
(webkitWebViewCreatePage):
(webkit_web_view_get_display):
(webkit_web_view_get_wpe_view):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.h.in:
* Source/WebKit/UIProcess/API/wpe/APIViewClient.h:
(API::ViewClient::enterFullScreen):
(API::ViewClient::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::wpeView const):
(WebKit::PageClientImpl::hostFileDescriptor):
(WebKit::PageClientImpl::setCursor):
(WebKit::PageClientImpl::setCursorHiddenUntilMouseMoves):
(WebKit::PageClientImpl::enterAcceleratedCompositingMode):
(WebKit::PageClientImpl::exitAcceleratedCompositingMode):
(WebKit::PageClientImpl::updateAcceleratedCompositingMode):
(WebKit::PageClientImpl::enterFullScreen):
(WebKit::PageClientImpl::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::View::View):
(WKWPE::m_wpeDisplay):
(WKWPE::View::~View):
(WKWPE::View::isFullScreen const):
(WKWPE::View::willEnterFullScreen):
(WKWPE::View::enterFullScreen):
(WKWPE::View::didEnterFullScreen):
(WKWPE::View::willExitFullScreen):
(WKWPE::View::exitFullScreen):
(WKWPE::View::didExitFullScreen):
(WKWPE::View::requestExitFullScreen):
(WKWPE::View::setFullScreen):
(WKWPE::View::updateAcceleratedSurface):
(WKWPE::View::touchPointsForEvent):
(WKWPE::View::setCursor):
(WKWPE::m_backend): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
(WKWPE::View::create):
(WKWPE::View::wpeView const):
(WKWPE::View::isFullScreen): Deleted.
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewClient.h:
* Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp:
(webkit_web_view_new):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
(WebKit::WebPageProxy::updateCurrentModifierState):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
(WebKit::DisplayVBlankMonitorDRM::create):
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::platformInitializeWebProcess):
* Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.cpp: Added.
(WebKit::AcceleratedBackingStoreDMABuf::create):
(WebKit::AcceleratedBackingStoreDMABuf::AcceleratedBackingStoreDMABuf):
(WebKit::AcceleratedBackingStoreDMABuf::~AcceleratedBackingStoreDMABuf):
(WebKit::AcceleratedBackingStoreDMABuf::updateSurfaceID):
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBuffer):
(WebKit::AcceleratedBackingStoreDMABuf::didCreateBufferSHM):
(WebKit::AcceleratedBackingStoreDMABuf::didDestroyBuffer):
(WebKit::AcceleratedBackingStoreDMABuf::frame):
(WebKit::AcceleratedBackingStoreDMABuf::frameDone):
(WebKit::AcceleratedBackingStoreDMABuf::bufferRendered):
* Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.h: Added.
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::wpeView const):
(WebKit::WebPageProxy::preferredBufferFormats const):
(WebKit::WebPageProxy::preferredBufferFormatsDidChange):
(WebKit::WebPageProxy::currentStateOfModifierKeys):
* Source/WebKit/WPEPlatform: Added.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.h:
(WebKit::AcceleratedSurface::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::SwapChain::setupBufferFormat):
(WebKit::AcceleratedSurfaceDMABuf::SwapChain::nextTarget):
(WebKit::AcceleratedSurfaceDMABuf::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:
* Source/WebKit/WebProcess/WebPage/wpe/WebPageWPE.cpp:
(WebKit::WebPage::preferredBufferFormatsDidChange):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::platformInitializeWebProcess):
* Source/WebKit/wpe/wpe-webkit-uninstalled.pc.in:
* Source/WebKit/wpe/wpe-webkit.pc.in:
* Source/cmake/FindLibInput.cmake: Added.
* Source/cmake/FindUdev.cmake: Added.
* Source/cmake/OptionsWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/MiniBrowser/wpe/main.cpp:
(createViewBackend):
(createWebView):
(activate):
(main):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.cmd_line):
* Tools/Scripts/webkitpy/port/wpe.py:
(WPEPort.setup_environ_for_server):
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_identifier_name_in_declaration):
* Tools/TestWebKitAPI/wpe/PlatformWebViewWPE.cpp:
(TestWebKitAPI::PlatformWebView::initialize):
* Tools/WebKitTestRunner/EventSenderProxy.h:
* Tools/WebKitTestRunner/Options.cpp:
(WTR::handleOptionWPEPlatformAPI):
(WTR::OptionsHandler::OptionsHandler):
* Tools/WebKitTestRunner/Options.h:
* Tools/WebKitTestRunner/PlatformWPE.cmake:
* Tools/WebKitTestRunner/PlatformWebView.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::initialize):
* Tools/WebKitTestRunner/TestController.h:
(WTR::TestController::useWPEPlatformAPI const):
* Tools/WebKitTestRunner/libwpe/EventSenderProxyClient.h: Copied from Source/WebKit/UIProcess/API/wpe/APIViewClient.h.
(WTR::EventSenderProxyClient::addTouchPoint):
(WTR::EventSenderProxyClient::updateTouchPoint):
(WTR::EventSenderProxyClient::touchStart):
(WTR::EventSenderProxyClient::touchMove):
(WTR::EventSenderProxyClient::touchEnd):
(WTR::EventSenderProxyClient::touchCancel):
(WTR::EventSenderProxyClient::clearTouchPoints):
(WTR::EventSenderProxyClient::releaseTouchPoint):
(WTR::EventSenderProxyClient::cancelTouchPoint):
(WTR::EventSenderProxyClient::EventSenderProxyClient):
* Tools/WebKitTestRunner/libwpe/EventSenderProxyClientLibWPE.cpp: Copied from Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp.
(WTR::EventSenderProxyClientLibWPE::EventSenderProxyClientLibWPE):
(WTR::viewBackend):
(WTR::secToMsTimestamp):
(WTR::senderButtonToWPEButton):
(WTR::modifierForButton):
(WTR::wkEventModifiersToWPE):
(WTR::EventSenderProxyClientLibWPE::mouseDown):
(WTR::EventSenderProxyClientLibWPE::mouseUp):
(WTR::EventSenderProxyClientLibWPE::mouseMoveTo):
(WTR::EventSenderProxyClientLibWPE::mouseScrollBy):
(WTR::wpeKeySymForKeyRef):
(WTR::EventSenderProxyClientLibWPE::keyDown):
(WTR::EventSenderProxyClientLibWPE::addTouchPoint):
(WTR::EventSenderProxyClientLibWPE::updateTouchPoint):
(WTR::EventSenderProxyClientLibWPE::touchStart):
(WTR::EventSenderProxyClientLibWPE::touchMove):
(WTR::EventSenderProxyClientLibWPE::touchEnd):
(WTR::EventSenderProxyClientLibWPE::touchCancel):
(WTR::EventSenderProxyClientLibWPE::clearTouchPoints):
(WTR::EventSenderProxyClientLibWPE::releaseTouchPoint):
(WTR::EventSenderProxyClientLibWPE::cancelTouchPoint):
(WTR::EventSenderProxyClientLibWPE::getUpdatedTouchEvents):
(WTR::EventSenderProxyClientLibWPE::removeUpdatedTouchEvents):
(WTR::EventSenderProxyClientLibWPE::prepareAndDispatchTouchEvent):
* Tools/WebKitTestRunner/libwpe/EventSenderProxyClientLibWPE.h: Added.
* Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp:
(WTR::EventSenderProxy::EventSenderProxy):
(WTR::EventSenderProxy::mouseDown):
(WTR::EventSenderProxy::mouseUp):
(WTR::EventSenderProxy::mouseMoveTo):
(WTR::EventSenderProxy::mouseScrollBy):
(WTR::EventSenderProxy::keyDown):
(WTR::EventSenderProxy::addTouchPoint):
(WTR::EventSenderProxy::updateTouchPoint):
(WTR::EventSenderProxy::touchStart):
(WTR::EventSenderProxy::touchMove):
(WTR::EventSenderProxy::touchEnd):
(WTR::EventSenderProxy::touchCancel):
(WTR::EventSenderProxy::clearTouchPoints):
(WTR::EventSenderProxy::releaseTouchPoint):
(WTR::EventSenderProxy::cancelTouchPoint):
(WTR::viewBackend): Deleted.
(WTR::secToMsTimestamp): Deleted.
(WTR::senderButtonToWPEButton): Deleted.
(WTR::modifierForButton): Deleted.
(WTR::wkEventModifiersToWPE): Deleted.
(WTR::wpeKeySymForKeyRef): Deleted.
(WTR::EventSenderProxy::getUpdatedTouchEvents): Deleted.
(WTR::EventSenderProxy::removeUpdatedTouchEvents): Deleted.
(WTR::EventSenderProxy::prepareAndDispatchTouchEvent): Deleted.
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClient.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
(WTR::PlatformWebViewClient::view const):
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClientLibWPE.cpp: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.cpp.
(WTR::PlatformWebViewClientLibWPE::PlatformWebViewClientLibWPE):
(WTR::PlatformWebViewClientLibWPE::~PlatformWebViewClientLibWPE):
(WTR::PlatformWebViewClientLibWPE::addToWindow):
(WTR::PlatformWebViewClientLibWPE::removeFromWindow):
(WTR::PlatformWebViewClientLibWPE::snapshot):
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClientLibWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
* Tools/WebKitTestRunner/libwpe/PlatformWebViewLibWPE.cpp:
(WTR::PlatformWebView::PlatformWebView):
(WTR::PlatformWebView::addToWindow):
(WTR::PlatformWebView::removeFromWindow):
(WTR::PlatformWebView::windowSnapshotImage):
* Tools/WebKitTestRunner/wpe/EventSenderProxyClientWPE.cpp: Added.
(WTR::EventSenderProxyClientWPE::EventSenderProxyClientWPE):
(WTR::secToMsTimestamp):
(WTR::wkEventModifiersToWPE):
(WTR::eventSenderButtonToWPEButton):
(WTR::modifierForButton):
(WTR::EventSenderProxyClientWPE::mouseDown):
(WTR::EventSenderProxyClientWPE::mouseUp):
(WTR::EventSenderProxyClientWPE::mouseMoveTo):
(WTR::EventSenderProxyClientWPE::mouseScrollBy):
(WTR::wpeKeyvalForKeyRef):
(WTR::EventSenderProxyClientWPE::keyDown):
* Tools/WebKitTestRunner/wpe/EventSenderProxyClientWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.
* Tools/WebKitTestRunner/wpe/PlatformWebViewClientWPE.cpp: Added.
(WTR::PlatformWebViewClientWPE::PlatformWebViewClientWPE):
(WTR::PlatformWebViewClientWPE::~PlatformWebViewClientWPE):
(WTR::PlatformWebViewClientWPE::addToWindow):
(WTR::PlatformWebViewClientWPE::removeFromWindow):
(WTR::PlatformWebViewClientWPE::snapshot):
* Tools/WebKitTestRunner/wpe/PlatformWebViewClientWPE.h: Copied from Source/WebKit/UIProcess/API/C/wpe/WKView.h.

Canonical link: https://commits.webkit.org/271330@main
